src/Build/Graph/ProjectInterpretation.cs

```diff
@@ -161,11 +165,15 @@ public IEnumerable<ReferenceInfo> GetReferences(ProjectInstance requesterInstanc
                             referenceGlobalProperties.Remove(ConfigurationMetadataName);
                             referenceGlobalProperties.Remove(PlatformMetadataName);
                         }
+                        else
+                        {
+                            configurationDefined = true;
+                        }
                     }
                 }
 
                 // Note: Dynamic platform resolution is not enabled for sln-based builds.
-                else if (!projectReferenceItem.HasMetadata(SetPlatformMetadataName) && enableDynamicPlatformResolution)
+                if (!projectReferenceItem.HasMetadata(SetPlatformMetadataName) && enableDynamicPlatformResolution && !configurationDefined)
```

Comment:
extreme nit: prefer the cheapest and most likely to fail checks in short-circuiting comparisons. Here there are a couple of bools which cost nothing to check, and a method call which is comparatively more expensive. It's not a very expensive method so this doesn't change much, but it's an easy refactor:

```suggestion
                if (enableDynamicPlatformResolution && !configurationDefined && !projectReferenceItem.HasMetadata(SetPlatformMetadataName))
```